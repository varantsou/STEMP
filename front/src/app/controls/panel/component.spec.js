import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Panel from './component';

const props = {
    className: 'test-class',
    title: 'Test title',
    buttonText: 'Edit',
    onClick: jest.fn()
};

describe('Should render Panel component', () => {
    test('with properties', () => {
        const component = shallow(<Panel {...props} />);
        const output = shallowToJson(component);

        expect(output).toMatchSnapshot();
    });

    test('without properties', () => {
        const component = shallow(<Panel />);
        const output = shallowToJson(component);

        expect(output).toMatchSnapshot();
    });
});

describe('Panel component', () => {
    describe('Should implement "handleButtonClick" method', () => {
        test('with "onClick" property', () => {
            const component = shallow(<Panel {...props} />);
            const instance = component.instance();

            jest.spyOn(instance.props, 'onClick')
                .mockImplementation(() => {});

            instance.handleButtonClick();

            expect(instance.props.onClick).toBeCalled();
        });

        test('without "handleButtonClick" property', () => {
            const component = shallow(<Panel />);
            const instance = component.instance();

            expect(() => {
                instance.handleButtonClick();
            }).not.toThrow();
        });
    });
});
