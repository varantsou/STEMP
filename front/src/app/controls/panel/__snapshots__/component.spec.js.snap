// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Panel component with properties 1`] = `
<div
  className="panel test-class"
>
  <div
    className="panel__header"
  >
    <h2
      className="panel__title"
    >
      Test title
    </h2>
    <Button
      className="panel__button"
      onClick={[Function]}
      theme="link"
    >
      Edit
    </Button>
  </div>
</div>
`;

exports[`Should render Panel component without properties 1`] = `
<div
  className="panel"
>
  <div
    className="panel__header"
  />
</div>
`;
